### ISSUE TITLE
warn when 'Send Max' doesn't send everything (frozen coins / addrs)

### ISSUE THREAD

**Issue by markblundeberg** [Electron-Cash/Electron-Cash#1012]
Sometimes when I do 'Send Max' I really want to empty out my wallet-- and then afterwards my wallet balance isn't zero since some coins were left frozen in the wallet. Would be nice if in the Send tab a warning could appear (perhaps conditionally) saying "not including frozen X BCH" or something like that.

(This has especially been happening with cash shuffle leaving some coins frozen, but in general it applies.)

**Comment by cculianu** [comment]
Yes -- I wanted this too but I never actually went through with thinking about how to fix it.

And you are right, this is a useful feature especially for Cash Shuffle. 

I will add messaging to the send tab in the case where Max really isn't max because of Cash Shuffle.

This reminds me -- there is a potential 'race condition' with respect to cash shuffle.  The 'Max' button marks the amount as being '!' internally and when the coin choose and other data structures see '!' they then take ALL THE COINS.

Since cash shuffle freezes and unfreezes coins behind the scenes while the user is doing stuff -- potentially the user hits max, sees amount XXX, and later when they actually send the tx they get amount YYY (which may be greater than, equal to, or less than XXX -- no guarantees either way!!).

This is potentially a hugely annoying issue!!

I'll open up an issue separately in the CashShuffle repo about that.

But yes -- UI messaging when Max isn't really max is useful in general.  I'll work on that.
